erpnext/controllers/accounts_controller.py

```diff
@@ -910,10 +914,16 @@ def get_value_in_transaction_currency(self, account_currency, args, field):
 			return flt(args.get(field, 0) / self.get("conversion_rate", 1))
 
 	def validate_qty_is_not_zero(self):
-		if self.doctype != "Purchase Receipt":
-			for item in self.items:
-				if not item.qty:
-					frappe.throw(_("Item quantity can not be zero"))
+		for item in self.items:
+			if self.doctype == "Purchase Receipt" and item.rejected_qty:
+				continue
```

Comment:
Instead, we might want to move special cases such as this into BuyingController or even PurchaseReceipt.

In this case we’d have to change the logic a bit, splitting out the iterated logic to a separate function that may then be overwritten or prepended with an additional check in a child class.

If this is preferred, I’d mark the PR a draft again and try to come up with a good way making more use of inheritance before bringing the PR back to review.

.validate_qty_is_not_zero(self)`,
then call `super.validate_qty_is_not_zero(self)`?

Or even into a `function PurchaseReceipt.validate_qty_is_not_zero()`?